### PR TITLE
jakarta libs upgrade

### PR DESCRIPTION
**Pull request by lukasz-wolski**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSRD-72

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -158,14 +158,18 @@ idea {
...
-            details.useVersion "4.1.77.Final"
-        }
-        if (details.requested.group == 'com.ibm.icu') {
-            details.useVersion "66.1"
-        }
-    }
+//    resolutionStrategy.eachDependency { details ->
+//        if (details.requested.group == 'io.netty') {
+//            details.useVersion "4.1.77.Final"
+//        }
```
Comment (kiran-yenigala-hmcts) [inline build.gradle]: please delete commented code

File: build.gradle [hunk 2]
```diff
@@ -308,16 +311,17 @@ dependencies {
     implementation group: 'org.springframework', name: 'spring-webmvc', version: versions.springVersion
 
     implementation group: 'com.github.ben-manes.caffeine', name: 'caffeine', version: '3.0.4'
-    implementation (group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.13.2.1') {
-        force = true
-    }
-    implementation (group: 'com.fasterxml.jackson.core', name: 'jackson-core', version: versions.jackson) {
-        force = true
-    }
+    implementation (group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.15.2')
```
Comment (kiran-yenigala-hmcts) [inline build.gradle]: can't we use 2.13.2 ( version: versions.jackson))

File: build.gradle [hunk 3]
```diff
@@ -332,58 +336,64 @@ dependencies {
...
-    implementation ('com.github.hmcts:service-auth-provider-java-client:4.0.3') {
+    implementation ('com.github.hmcts:service-auth-provider-java-client:5.1.0') {
         exclude group: 'org.springframework.cloud', module: 'spring-cloud-openfeign-core'
     }
-    implementation group: 'org.springframework.security', name: 'spring-security-crypto', version: '5.7.1'
+    implementation group: 'org.springframework.security', name: 'spring-security-crypto', version: '5.8.5'
     implementation 'net.logstash.logback:logstash-logback-encoder:6.4'
 
-    implementation group: 'com.microsoft.azure', name: 'azure-storage-blob', version: '11.0.0'
+    implementation group: 'com.azure', name: 'azure-storage-blob', version: '12.18.0'
```
Comment (kiran-yenigala-hmcts) [inline build.gradle]: latest is 12.23.0, better to use latest version, if it is compatable

**Review by kiran-yenigala-hmcts — COMMENTED** [review]
Minor comments

File: build.gradle (hunk 2 above)
Comment (lukasz-wolski) [reply build.gradle]: local dependency check scan flags jackson-databind 2.13.2 as vulnerable to CVE-2020-36518, CVE-2022-42003, CVE-2022-42004

**Review by kiran-yenigala-hmcts — APPROVED** [review]

**Pull request merged**